apps/battery/app.js

```diff
@@ -1,163 +1,153 @@
 'use strict'
-var logger = require('logger')('BATTERY')
+
+var Application = require('@yodaos/application').Application
+var SpeechSynthesis = require('@yodaos/speech-synthesis').SpeechSynthesis
+
+var logger = require('logger')('@battery')
 var util = require('util')
+var path = require('path')
 var prop = require('@yoda/property')
-var PROP_KEY = 'rokid.battery10.times'
-var TEMPERATURE_LIGHT_RES = 'system://temperatureBattery.js'
 var battery = require('@yoda/battery')
+var _ = require('@yoda/util')._
+var MediaPlayer = require('@yoda/multimedia').MediaPlayer
+
 var Const = require('./constant.json')
+
+var PROP_KEY = 'rokid.battery10.times'
+var TEMPERATURE_LIGHT_RES = 'system://temperatureBattery.js'
 var constant = Const.constant
 var resourcePath = Const.resource
 
-module.exports = function (activity) {
-  var STRING_NOBATTERY = '当前产品没有电池，使用期间请连接电源'
-  activity.media.on('error', (error) => {
-    logger.warn(error)
-  })
+function queryBatteryStatus () {
+  return battery.getBatteryInfo()
+}
 
-  function withoutBattery () {
-    speakAndExit(STRING_NOBATTERY)
+function parseTime (time) {
+  if (time < 0) {
+    return {
+      hour: 0,
+      minute: 0
+    }
   }
-
-  function queryBatteryStatus () {
-    return battery.getBatteryInfo()
+  var h = Math.floor(time / 60)
+  var m = time % 60
+  if (h < 0) {
+    h = 0
   }
-
-  function speakAndExit (text) {
-    return activity.tts.speak(text)
-      .then(() => {
-        logger.log('speakAndExit end will exit')
-        activity.exit()
-      })
+  return {
+    hour: h,
+    minute: m
   }
+}
 
-  function parseTime (time) {
-    if (time < 0) {
-      return {
-        hour: 0,
-        minute: 0
-      }
+function playMediaAsync (url) {
+  return new Promise((resolve, reject) => {
+    if (!path.isAbsolute(url)) {
+      url = path.resolve(url)
     }
-    var h = Math.floor(time / 60)
-    var m = time % 60
-    if (h < 0) {
-      h = 0
-    }
-    return {
-      hour: h,
-      minute: m
+    logger.log('notify media:', url)
+    var player = new MediaPlayer()
+    player.start(url)
+    player.on('playbackcomplete', resolve)
+    player.on('error', reject)
+  })
+}
+
+function lowerPower (percent, isPlaying) {
+  var url
+  if (percent === 10) {
+    url = resourcePath.lowPower10
+  } else if (percent === 20) {
+    if (isPlaying && isPlaying === 'true') {
```

Comment:
No need for 1st check:

```suggestion
    if (isPlaying === 'true') {
```